app/controllers/admin/cases_controller.rb

```diff
@@ -59,77 +59,50 @@ def count_cases_created_on(date)
 
   def select_type
     permitted_correspondence_types
+
     render :select_type
   end
 
   def prepare_new_foi
-    @correspondence_type_key = params[:correspondence_type]
-
-    case_creator = CTS::Cases::Create.new(Rails.logger,
-                                          type: 'Case::FOI::Standard' )
-    @case = case_creator.new_case
-    @case.responding_team = BusinessUnit.responding.responding_for_correspondence_type(CorrespondenceType.foi).active.sample
-    @case.flag_for_disclosure_specialists = 'no'
-    @target_states = available_target_states
-    @selected_state = 'drafting'
-    @s3_direct_post = S3Uploader.s3_direct_post_for_case(@case, 'requests')
+    prepare_new_case('Case::FOI::Standard', CorrespondenceType.foi)
 
     render :new
   end
 
   def prepare_new_sar
-    @correspondence_type_key = params[:correspondence_type]
-
-    case_creator = CTS::Cases::Create.new(Rails.logger, type: 'Case::SAR' )
-    @case = case_creator.new_case
-    @case.responding_team = BusinessUnit.responding.responding_for_correspondence_type(CorrespondenceType.sar).active.sample
-    @target_states = available_target_states
-    @selected_state = 'drafting'
-    @s3_direct_post = S3Uploader.s3_direct_post_for_case(@case, 'requests')
+    prepare_new_case('Case::SAR', CorrespondenceType.sar)
     @case.reply_method = 'send_by_email'
 
     render :new
   end
 
   def prepare_new_ico
-    @correspondence_type_key = params[:correspondence_type]
-    case_creator = CTS::Cases::Create.new(Rails.logger, type: 'Case::ICO::FOI' )
-    @case = case_creator.new_case
-    @case.responding_team = BusinessUnit.responding.responding_for_correspondence_type(CorrespondenceType.ico).active.sample
-    @target_states = available_target_states
-    @selected_state = 'drafting'
-    @case.ico_decision = Case::ICO::Base.ico_decisions.keys.sample
+    prepare_new_case('Case::ICO::FOI', CorrespondenceType.foi)
     @s3_direct_post = S3Uploader.s3_direct_post_for_case(@case, 'requests')
```

Comment:
I think this can also be removed since it's done on line 108 in `prepare_new_case`. Which means ... is there any custom code left for different case types at all? Or can we have the controller call `prepare_new_case` with the type as an argument, and a method that returns the class `Case::ICO::FOI` and original type ...